src/Streamly/Internal/Data/Parser/ParserD/Type.hs

```diff
@@ -438,7 +438,7 @@ noErrorUnsafeSplitWith func (Parser stepL initialL extractL)
         r <- stepR st a
         return $ case r of
             Partial n s -> Partial n (SeqParseR f s)
-            Continue n s -> Partial n (SeqParseR f s)
+            Continue n s -> Continue n (SeqParseR f s)
```

Comment:
This should be `Partial`. `noErrorUnsafeSplitWith` isf or parsers that don't fail.